dnf-behave-tests/features/builddep.feature

```diff
@@ -80,3 +80,62 @@ Scenario: Builddep on SPEC with non-available Source0
 
    Error: Some packages could not be found.
    """
+
+@bz1758459
+Scenario: I can exclude package version and install the best candidate
+  Given I use repository "dnf-ci-fedora"
+    And I enable plugin "builddep"
+  Given I create file "dummy-pkg.spec" with
+   """
+   Name: dummy-pkg
+   Summary: dummy-pkg summary
+   Version: 1.0
+   Release: 1
+   License: GPL
+   BuildRequires: nodejs
+   %description
+   This is a dummy-pkg description
+   %build
+   %files
+   %changelog
+   """
+   When I execute dnf with args "builddep {context.dnf.installroot}/dummy-pkg.spec --exclude nodejs-1:5.12.1-1.fc29 --best"
+   Then the exit code is 0
+    And Transaction contains
+        | Action                | Package                       |
+        | install               | nodejs-1:5.11.1-1.fc29.x86_64 |
+    And stderr does not contain "Traceback"
+
+@bz1758459
+Scenario: I can exclude package version and install the best candidate with modules
+  Given I use repository "dnf-ci-fedora-modular"
+    And I use repository "dnf-ci-fedora"
+    And I enable plugin "builddep"
+  Given I create file "dummy-pkg.spec" with
+   """
+   Name: dummy-pkg
+   Summary: dummy-pkg summary
+   Version: 1.0
+   Release: 1
+   License: GPL
+   BuildRequires: nodejs
+   %description
+   This is a dummy-pkg description
+   %build
+   %files
+   %changelog
+   """
+   When I execute dnf with args "builddep {context.dnf.installroot}/dummy-pkg.spec --exclude nodejs-1:11.0.0-1.module_2311+8d497411.fc29 --best"
+   Then the exit code is 0
+    And Transaction contains
+        | Action                 | Package                                       |
+        | install                | setup-0:2.12.1-1.fc29.noarch                  |
+        | install                | filesystem-0:3.9-2.fc29.x86_64                |
+        | install                | basesystem-0:11-6.fc29.noarch                 |
+        | install                | glibc-all-langpacks-0:2.28-9.fc29.x86_64      |
+        | install                | glibc-common-0:2.28-9.fc29.x86_64             |
+        | install                | glibc-0:2.28-9.fc29.x86_64                    |
+        | install                | npm-1:8.11.4-1.module_2030+42747d40.x86_64    |
+        | install                | nodejs-1:8.11.4-1.module_2030+42747d40.x86_64 |
+    And stderr does not contain "Traceback"
+    And stderr does not contain "Modular dependency problem"
```

Comment:
You also probably shouldn't check for this, because in the specific situation of there being a modular dependency problem, the exit status probably also would be non-zero or at least the output would be different from what you expect? (not entirely sure about this but I think it is so)

Now the command could go haywire in such a way it would actually end with zero return code and the same output and still print the "Modular dependency problem", but then we're getting into the realm of it printing arbitrary errors which this check won't catch (because it's a negative check and that means it will pass on anything but the exact error string). So if you go this path, you end up with the conclusion that you need to check for exact error outputs (as opposed to a "contains" check).

So basically I think this check is redundant. Again, if you can, you should check for stderr being empty. Ultimately, we should be able to check for the exact content of stderr, whatever it is.